sdk/nodejs/runtime/invoke.ts

```diff
@@ -100,3 +128,136 @@ export async function invoke(tok: string, props: Inputs, opts?: InvokeOptions):
         done();
     }
 }
+
+function invokeSync(tok: string, props: any, opts: InvokeOptions): Promise<any> {
+    console.log("Performing synchronous invoke.");
+    const label = `Invoking function: tok=${tok} synchronously`;
+    log.debug(label +
+        excessiveDebugOutput ? `, props=${JSON.stringify(props)}` : ``);
+
+    const obj = gstruct.Struct.fromJavaScript(serializePropertiesSync(props));
+    log.debug(`Invoke RPC prepared: tok=${tok}` + excessiveDebugOutput ? `, obj=${JSON.stringify(obj)}` : ``);
+
+    // Fetch the sync monitor and make an RPC request.
+    const syncInvokes = getSyncInvokes();
```

Comment:
@pgavlin @lukehoban We need to make a decision on what to do here if `getSyncInvokes()` fails.  Specifically, it could fail if someone has an older CLI (with an older `pulumi-language-nodejs` plugin) but revs their `@pulumi/pulumi` version.

1. it's unclear if users update these independently.  Or if they just run our scripts to update their `pulumi` install, in which case this would not happen.
2. Presuming that it *does* happen, we have several options available as to what we can do.  Including:
    1. We can error here and require that they update their `pulumi` install.
    2. We can silently fallback to our existing behavior where we call the async codepath and then deasync it. 
    3. We can fallback to our existing behavior and issue a warning that there is a cli mismatch and they should upgrade.

Reply:
For now, i'm going with 2.iii.  i.e. we warn and fallback.  The user will see this on the command line:

![image](https://user-images.githubusercontent.com/4564579/66775159-3be72100-ee78-11e9-9d5b-85f3174d643c.png)